client/src/components/PostPreview/PostPreview.tsx

```diff
@@ -37,13 +43,18 @@ export default function PostPreview(props:PostPreviewProps) {
         <Link className="post-preview-title" to={`/posts/${key}`}>{title}</Link>
         
         {/* truncates the text if it is longer than PREVIEW_TEXT_LENGTH */}
-        <p className="post-preview-text">{text.substring(0, PREVIEW_TEXT_LENGTH)}{text.length > PREVIEW_TEXT_LENGTH && "..."}</p>
+        <p className="post-preview-text shadow-small">{text.substring(0, PREVIEW_TEXT_LENGTH)}{text.length > PREVIEW_TEXT_LENGTH && "..."}</p>
         
 
         <div className="post-preview-footer">
-            <div className="counter" onClick={handleLike}><FaHeart /><span>{likes}</span></div>
+            <div className="counter" onClick={handleLike}>
+                {isLiked ? <FaHeart style={{color: "#ff6b6b"}}/> : <FaRegHeart />}
+                <span>
+                    {isLiked ? likes + 1 : likes}
```

Comment:
If this object ever refreshes from the DB, you'll double count the 'like'. Also related, if you reload this page from scratch you need to know whether you liked the post on a previous visit, so that information would have to be encapsulated in the props.post most likely. Maybe a bit outside the scope of the project, but things you'd have to solve in a real situation.